wandb/sdk/wandb_artifacts.py

```diff
@@ -209,6 +222,63 @@ def add_reference(self, uri, name=None, checksum=True, max_objects=None):
         for entry in manifest_entries:
             self._manifest.add_entry(entry)
 
+        return manifest_entries
+
+    # TODO: name this add_obj?
+    def add(self, obj, name):
+        """adds `obj` to the artifact, located at `name`. You can use Artifact#get(`name`) after downloading
+        the artifact to retrieve this object.
+        
+        Arguments:
+        - `obj`:wandb.Media - the object to save
+        - `name`:str - the path to save
+        """
+        if isinstance(obj, Media):
+
+            # If the object is coming from another artifact, save it as a reference
+            if hasattr(obj, "_source") and obj._source is not None:
+                suffix = "." + obj.get_json_suffix() + ".json"
+                ref_path = obj._source["artifact"].get_path(
+                    obj._source["name"] + suffix
+                )
+                path = name + suffix
+                return self.add_reference(ref_path, path)[0]
+
+            # Otherwise, save the object directly via json
+            else:
+                obj_id = id(obj)
```

Comment:
You can only addd specific Media types to artifacts now. Should we detect if they try to add other types and give a nice error message?

Reply:
Incoming next commit